File: brjobs-java/src/main/java/ads/uninassau/brjobs/model/TipoUsuario.java
package ads.uninassau.brjobs.model;

/**
 * Enum que define os tipos de usuários no sistema.
 *
 * CONTRATANTE: Pessoa que contrata serviços
 * PRESTADOR: Pessoa que oferece serviços
 */
public enum TipoUsuario {
    CONTRATANTE("Contratante"),
    PRESTADOR("Prestador de Serviço");

    private final String descricao;

    TipoUsuario(String descricao) {
        this.descricao = descricao;
    }

    public String getDescricao() {
        return descricao;
    }

    /**
     * Verifica se o tipo é PRESTADOR
     */
    public boolean isPrestador() {
        return this == PRESTADOR;
    }

    /**
     * Verifica se o tipo é CONTRATANTE
     */
    public boolean isContratante() {
        return this == CONTRATANTE;
    }
}

File: brjobs-flutter/lib/features/auth/domain/tipo_usuario.dart
/// Tipo de usuário no backend (`TipoUsuario`).
enum TipoUsuario {
  contratante('CONTRATANTE'),
  prestador('PRESTADOR');

  const TipoUsuario(this.wire);

  /// Valor exato esperado/retornado pela API.
  final String wire;

  bool get isPrestador => this == TipoUsuario.prestador;
  bool get isContratante => this == TipoUsuario.contratante;

  static TipoUsuario? fromWire(String? value) {
    if (value == null) return null;
    final v = value.toUpperCase();
    for (final t in TipoUsuario.values) {
      if (t.wire == v) return t;
    }
    return null;
  }

  String get label => switch (this) {
        TipoUsuario.contratante => 'Contratante',
        TipoUsuario.prestador => 'Prestador de serviço',
      };
}
